crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -566,6 +566,20 @@ impl<'w> UnsafeWorldCell<'w> {
             .get(component_id)?
             .get_with_ticks()
     }
+
+    // Returns a mutable reference to the underlying world's [`CommandQueue`].
+    /// # Safety
+    /// It is the callers responsibility to ensure that
+    /// - the [`UnsafeWorldCell`] has permission to access the queue mutably
+    /// - no mutable references to the queue exist at the same time
+    pub(crate) unsafe fn get_command_queue(self) -> &'w mut CommandQueue {
+        // SAFETY:
+        // - caller ensures there are no existing mutable references
+        // - caller ensures that we have permission to access the queue
+        let ptr = unsafe { &mut (*self.0).command_queue as *mut _ };
```

Comment:
[`addr_of_mut`](https://doc.rust-lang.org/stable/std/ptr/macro.addr_of_mut.html) would be relevant here.